File: 258.AddDigits/Add.java
class Solution {
    public int addDigits(int num) {
        while (num >= 10) {
        	int tmp = num;
        	num = 0;
        	while (tmp > 0) {
        		num += tmp % 10;
        		tmp /= 10;
        	}
        }
        return num;
    }

    public int addDigits(int num) {
    	return 1 + (num - 1) % 9;
    }
}

File: 258.AddDigits/add.py
class Solution(object):
    def addDigits(self, num):
        """
        :type num: int
        :rtype: int
        """
        while num >= 10:
        	tmp = num
        	num = 0
        	while tmp:
        		num += tmp % 10
        		tmp /= 10
        return num

    def addDigits(self, num):
    	return 1 + (num - 1) % 9